crates/flux-infer/src/infer.rs

```diff
@@ -366,15 +366,14 @@ impl<'infcx, 'genv, 'tcx> InferCtxt<'infcx, 'genv, 'tcx> {
         &mut self,
         t: &Binder<T>,
         f: impl FnOnce(&mut InferCtxt<'_, 'genv, 'tcx>, T) -> U,
-    ) -> U
+    ) -> InferResult<U>
     where
         T: TypeFoldable,
     {
         self.ensure_resolved_evars(|infcx| {
             let t = t.replace_bound_refts_with(|sort, mode, _| infcx.fresh_infer_var(sort, mode));
             Ok(f(infcx, t))
         })
-        .unwrap()
```

Comment:
I prefer to keep the `unwrap` here or turning this into a `query_bug!` (i.e., `.map_err(|e| query_bug!(...))`. Otherwise, I feel the error message you get is confusing.